src/granite_orm.cr

```diff
@@ -131,15 +131,19 @@ class Granite::ORM
 
     # keep a hash of the params that will be passed to the adapter.
     def params
-      params = [] of DB::Any
+      parsed_params = [] of DB::Any
       {% for name, type in FIELDS %}
-        params << {{name.id}}
+        {% if type.id == Time.id %}
+          parsed_params << {{name.id}}.try(&.to_s("%F %X"))
+        {% else %}
+          parsed_params << {{name.id}}
+        {% end %}
       {% end %}
       {% if SETTINGS[:timestamps] %}
-        params << created_at.not_nil!.to_s("%F %X")
-        params << updated_at.not_nil!.to_s("%F %X")
+        parsed_params << created_at.not_nil!.to_s("%F %X")
+        parsed_params << updated_at.not_nil!.to_s("%F %X")
```

Comment:
@veelenga If nil wasn't possible then we couldn't initialize something without setting all values.
```cr
post = Post.new
post.title = "time stuff"
post.save # Time stamps should get set at save not initialize.
```